custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -353,8 +359,68 @@ def run_test_predict(self):
                 df = handle_missing_colnames(df)
             df.to_csv(__tempfile.name, index=False)
             self.options.input = __tempfile.name
+        if self.target_type != TargetType.TRANSFORM:
+            self._check_prediction_side_effects()
         self._run_fit_and_predictions_pipelines_in_mlpiper()
 
+    def _check_prediction_side_effects(self):
+        rtol = 2e-02
+        atol = 1e-06
+        input_extension = os.path.splitext(self.options.input)
+        is_sparse = input_extension[1] == ".mtx"
+
+        if is_sparse:
+            df = pd.DataFrame(mmread(self.options.input).tocsr())
+        else:
+            df = pd.read_csv(self.options.input)
+        samplesize = min(1000, max(int(len(df) * 0.1), 10))
+        data_subset = df.sample(n=samplesize, random_state=42)
+
+        if is_sparse:
+            __tempfile_sample = NamedTemporaryFile(suffix=".mtx")
+            sparse_mat = vstack(x[0] for x in data_subset.values)
+            mmwrite(__tempfile_sample.name, sparse_mat)
+        else:
+            __tempfile_sample = NamedTemporaryFile(suffix=".csv")
+            data_subset.to_csv(__tempfile_sample.name, index=False)
+
+        if self.target_type == TargetType.BINARY:
+            labels = [self.options.negative_class_label, self.options.positive_class_label]
+        elif self.target_type == TargetType.MULTICLASS:
+            labels = self.options.class_labels
+        else:
+            labels = None
+
+        with DrumServerRun(
+            self.target_type.value,
```

Comment:
for some reason the drum push tests are failing on jarvis here but not locally - the traceback seemingly corresponds to code from a previous commit? I dunno